tools/c7n_azure/c7n_azure/storage_utils.py

```diff
@@ -86,7 +88,14 @@ def get_storage_keys(storage_account_id):
     def get_storage_from_uri(storage_uri):
         parts = urlparse(storage_uri)
         storage_name = str(parts.netloc).partition('.')[0]
-        container_name = parts.path.partition('/')[2]
+
+        path_parts = os.path.split(parts.path)
+        container_name = path_parts[0].strip('/')
+        if len(path_parts) > 1:
+            key_prefix = path_parts[1].strip('/')
```

Comment:
If the user provides some blob uri like`https://myaccount.blob.core.windows.net/mycontainer/myblob/seg1/seg2`, would we ignore `seg1` and `seg2`?

Reply:
At the moment only planning to support a container, and custodian builds the directory structure after the container, but we can probably make it let them specify arbitrary virtual directories